### DISCUSSION TITLE
Alert Message send failed

### DISCUSSION THREAD

**Discussion by seaurching** [grafana/mimir#2873] [Help and support]
#### Describe the bug
The alert message send filed while rules number above 1. And it show 
```
level=error ts=2022-08-28T05:31:00.097461Z caller=dispatch.go:354 component=MultiTenantAlertmanager user=anonymous component=dispatcher insight=true component=dispatcher msg="Notify for alerts failed" num_alerts=2 err="invalid service state: Terminated, expected: Running"
```
#### Image
![mimir](https://user-images.githubusercontent.com/7349963/187059333-7a36edcd-9285-4a67-ae20-b084c5dcbb3d.png)

#### Version
```
Mimir, version 2.2.0 (branch: HEAD, revision: 65344e2ed)
  go version:       go1.17.8
  platform:         darwin/amd64
```

**Comment by dimitarvdimitrov** [comment]
the log message you posted - do you see this in grafana, in the ruler logs, or in the alertmanager logs?

**Reply by chriskuchin** [reply]
I have a similar error and I see mine within the mimir alertmanager logs

The alertmanager runs fine for a few minutes but then that log starts and just spams my logs and further alerts fail to send

**Reply by dimitarvdimitrov** [reply]
I can see this happening in our alertmanager. But this is usually preceeded by either deleting the alerting configuration for a tenant or resharding - adding or removing AM replicas or an AM replica restarting. Most importantly, it should only last a few seconds. 

In your cases how long does this "spamming" last - a few seconds, a minute, an hour? Until restart?

Also, what are the Alertmanager logs preceding this "spamming?"

**Reply by chriskuchin** [reply]
So I made a few adjustments to my config increasing replicas of the alertmanager and increasing the log level it was originally `warn` which mean preceding the problem there were no logs. 

It hasn't repeated since then

**Reply by dimitarvdimitrov** [reply]
what replication factor were you running with when it was happening @chriskuchin ? I mean `alertmanager.sharding_ring.replication_factor`

**Comment by chriskuchin** [comment]
Was running with a factor of 1 and adjusted to 2

**Reply by dimitarvdimitrov** [reply]
thanks. looks like it's an issue with `replication_factor == 1` we are tracking this in https://github.com/grafana/mimir/issues/2910

**Comment by chriskuchin** [comment]
Awesome thanks good to know
